docs/source/distribution_implementation.md

```diff
@@ -0,0 +1,543 @@
+distribution.
+
+The following snippet illustrates how to create a new `RandomVariable`:
+
+```python
+
+from aesara.tensor.var import TensorVariable
+from aesara.tensor.random.op import RandomVariable
+
+# Create your own `RandomVariable`...
+class BlahRV(RandomVariable):
+    name: str = "blah"
+
+    # Provide the number of (output) dimensions for this RV
+    # (e.g. `0` for a scalar, `1` for a vector, etc.)
+    ndim_supp: int = 0
+
+    # Provide the number of (input) dimensions for each parameter of the RV
+    # (e.g. if there's only one vector parameter, `[1]`; for two parameters,
+    # one a matrix and the other a scalar, `[2, 0]`; etc.)
+    ndims_params: List[int] = [0, 0]
+
+    # The NumPy/Aesara dtype for this RV (e.g. `"int32"`, `"int64"`).
+    # The standard in the library is `"int64"` for discrete variables
+    # and `"floatX"` for continuous variables
+    dtype: str = "floatX"
+
+    # A pretty text and LaTeX representation for the RV
+    _print_name: Tuple[str, str] = ("blah", "\\operatorname{blah}")
+
+    # If you want to add a custom signature and default values for the
+    # parameters, do it like this. Otherwise this can be left out.
+    def __call__(self, loc=0.0, scale=1.0, size=None, **kwargs) -> TensorVariable:
+        return super().__call__(loc, scale, size=size, **kwargs)
+
+    # This is the Python code that produces samples.  Its signature will always
+    # start with a NumPy `RandomState` object, then the distribution
+    # parameters, and, finally, the size.
+    #
+    # This is effectively the v4 replacement for `Distribution.random`.
+    @classmethod
+    def rng_fn(
+        cls,
+        rng: np.random.RandomState,
+        loc: np.ndarray,
+        scale: np.ndarray,
+        size: Tuple[int, ...],
+    ) -> np.ndarray:
+        return scipy.stats.blah.rvs(loc, scale, random_state=rng, size=size)
+
+# Create an instance of the `RandomVariable` `Op`...
+blah = BlahRV()
+
+```
+
+Some important things to keep in mind:
+
+1. Everything inside the `rng_fn` method is pure Python code (as are the inputs)
+   and should not make use of other `Aesara` symbolic ops. The random method
+   should make use of the `rng` which is a Numpy
+   {`Random State`}`numpy.random.RandomState`, so that samples are reproducible.
+1. The `size` argument (together with the inputs shape) is the only way for the
+   user to specify the `batch_shape` (~number of repetitions) of the
+   `RandomVariable`.
+   The `rng_fn` will have to take this into consideration for correct output.
+   `size` is the specification used by `Numpy` and `Scipy` and works like
+   PyMC3 `shape` for univariate distributions, but is different for multivariate
+   distributions. Unfortunately there is no general reference documenting how
+   `size` ought to work for multivariate distributions.
+   This [discussion](https://github.com/numpy/numpy/issues/17669) may be helpful
+   to get more context.
```

Comment:
```suggestion
1. The `size` argument (together with the inputs shape) are the one way to specify
    the dimensions (minus the last `ndim_supp`) at the `RandomVariable` API level.
    At the user API level more options (`shape`, `dims`) are available, but this is taken
    care of by {`class`}`~pymc3.distributions.Distribution`.
   The `rng_fn` will have to take `size` into consideration for correct output.
   `size` is the specification used by `Numpy` and `Scipy` and works like
   PyMC3 `shape` for univariate distributions, but is different for multivariate
   distributions. Unfortunately there is no general reference documenting how
   `size` ought to work for multivariate distributions.
   This [discussion](https://github.com/numpy/numpy/issues/17669) may be helpful
   to get more context.
```

Reply:
Added a slightly different note on this below.